tests/UnitTests/Rsp.QuestionSetService.UnitTests/Services/QuestionServiceTests/GetQuestionByVersionTests.cs

```diff
@@ -0,0 +1,111 @@
+﻿using Rsp.QuestionSetService.Application.Contracts.Repositories;
+using Rsp.QuestionSetService.Application.Specifications;
+using Rsp.QuestionSetService.Domain.Entities;
+using Rsp.QuestionSetService.Services;
+
+namespace Rsp.QuestionSetService.UnitTests.Services.QuestionServiceTests;
+
+public class GetQuestionByVersionTests : TestServiceBase<QuestionService>
+{
+    [Fact]
+    public async Task GetQuestionsByVersion_ShouldReturnAllQuestions_ForTheGivenVersion()
+    {
+        // Arrange
+        var questions = new List<Question>()
+        {
+            new() { QuestionId = "1", QuestionText = "Question 1", VersionId = "v1" },
+            new() { QuestionId = "2", QuestionText = "Question 2", VersionId = "v1" },
+            new() { QuestionId = "3", QuestionText = "Question 3", VersionId = "v2" }
+        };
+
+        Mocker
+            .GetMock<IQuestionRepository>()
+            .Setup(repo => repo.GetQuestions(It.IsAny<QuestionSpecification>()))
+            .ReturnsAsync(questions.Where(q => q.VersionId == "v1"));
+
+        // Act
+        var result = await Sut.GetQuestionsByVersion("v1");
+
+        // Assert
+        result.ShouldNotBeNull();
+        result.Count().ShouldBe(questions.Where(q => q.VersionId == "v1").Count());
+
+        // Verify
+        Mocker
+            .GetMock<IQuestionRepository>()
+            .Verify(repo => repo.GetQuestions(It.IsAny<QuestionSpecification>()), Times.Once);
+    }
+
+    [Fact]
+    public async Task GetQuestionsByVersion_ShouldReturnEmptyList_WhenNoQuestionsExist_ForTheGivenVersion()
+    {
+        // Arrange
+        Mocker
+            .GetMock<IQuestionRepository>()
+            .Setup(repo => repo.GetQuestions(It.IsAny<QuestionSpecification>()))
+            .ReturnsAsync([]);
+
+        // Act
+        var result = await Sut.GetQuestionsByVersion("v1");
+
+        // Assert
+        result.ShouldNotBeNull();
+        result.ShouldBeEmpty();
```

Comment:
You can chain the methods
```suggestion
        result
            .ShouldNotBeNull();
            .ShouldBeEmpty();
```